### ISSUE TITLE
chore: useful reth / alloy type conversions

### ISSUE THREAD

**Issue by merklefruit** [paradigmxyz/op-rs#82]
The following traits should be implemented in reth to remove manual casting functions in op-rs:

- `impl From<reth::primitives::SealedBlock> for alloy::rpc::types::Block`
- `impl From<reth::primitives::SealedHeader> for alloy::consensus::Header`
- `impl From<reth::primitives::TransactionSigned> for Option<alloy::consensus::TxEnvelope>` (TryFrom is fine too)

**Comment by GrapeBaBa** [comment]
@merklefruit let me try

**Comment by GrapeBaBa** [comment]
@merklefruit When I looked [alloy](), it did't depend on `reth-primitives` and i thought it should not, so i am confused does this issue make sense?

**Comment by merklefruit** [comment]
Yeah that's the issue. We can just wait to see if Reth is interested in using Alloy types for this and then it will be possible. 
For now this is blocked - but would like to keep this open just to remember which type conversions we need eventually.
